From 655c4c72a3467037abd51aab29f0300e97caf54c Mon Sep 17 00:00:00 2001
From: Stepan Broz <sbroz@redhat.com>
Date: Thu, 25 Jul 2019 11:00:47 +0200
Subject: [PATCH] rpmpgp: Handle EOF without EOL better at END PGP

---
 rpmio/rpmpgp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/rpmio/rpmpgp.c b/rpmio/rpmpgp.c
index 061751a4d..46cd0f31a 100644
--- a/rpmio/rpmpgp.c
+++ b/rpmio/rpmpgp.c
@@ -1289,9 +1289,10 @@ static pgpArmor decodePkts(uint8_t *b, uint8_t **pkt, size_t *pktlen)
 		goto exit;
 	    }
 	    t += (sizeof("-----")-1);
-	    if (t >= te) continue;
+	    /* Handle EOF without EOL here, *t == '\0' at EOF */
+	    if (*t && (t >= te)) continue;
 	    /* XXX permitting \r here is not RFC-2440 compliant <shrug> */
-	    if (!(*t == '\n' || *t == '\r')) continue;
+	    if (!(*t == '\n' || *t == '\r' || *t == '\0')) continue;
 
 	    crcdec = NULL;
 	    crclen = 0;
-- 
2.26.2

